commons/zenoh-protocol-core/src/whatami.rs

```diff
@@ -259,12 +259,12 @@ impl BitOr for WhatAmIMatcher {
 impl BitOr for WhatAmI {
     type Output = WhatAmIMatcher;
     fn bitor(self, rhs: Self) -> Self::Output {
-        WhatAmIMatcher(unsafe { NonZeroU8::new_unchecked(self as u8 | rhs as u8) })
+        WhatAmIMatcher(unsafe { NonZeroU8::new_unchecked((self as u8 | rhs as u8) + 128) })
```

Comment:
I'd prefer `|` to `+`, as there's no risk of accidental carries ruining our day here